sets.go

```diff
@@ -27,24 +27,18 @@ func NewSet[T comparable](hasher Hasher[T], values ...T) Set[T] {
 //
 // This function will return a new set even if the set already contains the value.
 func (s Set[T]) Set(values ...T) Set[T] {
-	n := Set[T]{
-		m: s.m.clone(),
-	}
 	for _, value := range values {
-		n.m.set(value, struct{}{}, true)
+		s.m = s.m.Set(value, struct{}{})
```

Comment:
The idea of adding the varargs was to provide a convenient way to add many items, while avoiding creation of a new set for each value in the varargs. 

So I'd still use the `mutable=true` but initialise the new set differently. Maybe using `s.Items()` as above.

Reply:
This is not possible without copying the full set (or without making the implementation more complex, as I mentioned in the issue).
Copying the full set would be unexpected behaviour to me.